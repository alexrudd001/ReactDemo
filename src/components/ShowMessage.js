import React from 'react';

const ShowMessage = ({ messageType }) => {
    const errorMessage = 'Service is unavailable. Please try after sometime.';
    const successMessage = 'Request completed successfully.';
    return (
        <div>
            {messageType === 'success' && <p className="success"> {successMessage} </p>}
            {messageType === 'error' && <p className="error"> {errorMessage} </p>}
        </div>
    );
};

export default ShowMessage;
