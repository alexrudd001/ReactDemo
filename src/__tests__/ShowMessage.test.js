import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ShowMessage from '../components/ShowMessage';

test('renders ShowMessage correctly', () => {
    const wrapper = shallow(<ShowMessage />);
    expect(toJson(wrapper)).toMatchSnapshot();
});

test('renders ShowMessage correctly for success', () => {
    const wrapper = shallow(<ShowMessage messageType={'success'}/>);
    expect(toJson(wrapper)).toMatchSnapshot();
});

test('renders ShowMessage correctly for error', () => {
    const wrapper = shallow(<ShowMessage messageType={'error'}/>);
    expect(toJson(wrapper)).toMatchSnapshot();
});
