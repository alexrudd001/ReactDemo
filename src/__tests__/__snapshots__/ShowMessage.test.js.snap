// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ShowMessage correctly 1`] = `<div />`;

exports[`renders ShowMessage correctly for error 1`] = `
<div>
  <p
    className="error"
  >
     
    Service is unavailable. Please try after sometime.
     
  </p>
</div>
`;

exports[`renders ShowMessage correctly for success 1`] = `
<div>
  <p
    className="success"
  >
     
    Request completed successfully.
     
  </p>
</div>
`;
